backends/zypp/pk-backend-zypp.cpp

```diff
@@ -1454,6 +1454,20 @@ zypp_perform_execution (PkBackendJob *job, ZYpp::Ptr zypp, PerformType type, gbo
 		}
 
 		ResPool pool = ResPool::instance ();
+
+		// Products labeled as `dup` for update need to have solver switched to doUpgrade
+		if (type == UPDATE) {
+			for (ResPool::const_iterator it = pool.begin (); it != pool.end (); ++it) {
+				if ( it->isKind<Product>() ) {
+					static const Capability indicator( "product-update()", Rel::EQ, "dup" );
+					if ( it->operator->()->asKind<Product>()->referencePackage().provides().matches( indicator ) ) {
+						zypp->resolver()->setUpgradeMode(TRUE);
```

Comment:
Looks to me like `resolvePool ()` was already called at L1406, so switching the upgrade mode here will affect the next solver run, not the current one. Not sure if this is what you intended.

Reply:
But the transaction hasn't been run yet at this point, so shouldn't it affect the currently developing transaction?

Reply:
Not immediately.  Libzypp basically allows to manip pool items directly and to commit without running the solver. 

Changing `Resolver` flags or solver jobs does not immediately change the status of the items in the `ResPool`. You need to call `resolvePool()` to run the solver and apply the result to the pool items (i.e. updating their status).